controllers/watchers/target_pod.go

```diff
@@ -0,0 +1,563 @@
+						// The readiness probe is failing during the injection
+						if d.Disruption.Status.HasTarget(event.InvolvedObject.Name) {
+							eventsToSend[v1beta1.EventTargetReadinessProbeChangeDuringDisruption] = true
+						} else {
+							eventsToSend[v1beta1.EventTargetReadinessProbeChangeBeforeDisruption] = true
+						}
+					default:
+						eventsToSend[v1beta1.EventTargetPodWarningState] = true
+					}
+				} else {
+					eventsToSend[v1beta1.EventTargetPodWarningState] = true
+				}
+
+				d.Log.Debugw("warning event detected on target",
+					"target", targetName,
+					"reason", event.Reason,
+					"message", event.Message,
+					"timestamp", event.LastTimestamp.Time.Unix(),
+				)
+			case event.Reason == "Started":
+				if recoverTimestamp == nil {
+					recoverTimestamp = &event.LastTimestamp.Time
+				}
+
+				eventsToSend[v1beta1.EventTargetPodRecoveredState] = true
+
+				d.Log.Debugw("recovering event detected on target",
+					"target", targetName,
+					"reason", event.Reason,
+					"message", event.Message,
+					"timestamp", event.LastTimestamp.Time.Unix(),
+				)
+			case event.Reason == "Killing" && strings.Contains(event.Message, "Stopping container") && eventsToSend[v1beta1.EventTargetContainerWarningState]:
+				// this event indicates a safe killing of a container (can occur when we rollout or manually delete a pod for example)
+				// we remove the warning state event if it has been created when we compared the state of the containers
+				delete(eventsToSend, v1beta1.EventTargetContainerWarningState)
+			}
+		case "Node":
+			if event.Type == v1.EventTypeWarning {
+				eventsToSend[v1beta1.EventTargetNodeWarningState] = true
+
+				d.Log.Debugw("warning event detected on target",
+					"target", targetName,
+					"reason", event.Reason,
+					"message", event.Message,
+					"timestamp", event.LastTimestamp.Time.Unix(),
+				)
+			} else if event.Reason == "NodeReady" {
+				if recoverTimestamp == nil {
+					recoverTimestamp = &event.LastTimestamp.Time
+				}
+
+				eventsToSend[v1beta1.EventTargetNodeRecoveredState] = true
+
+				d.Log.Debugw("recovering event detected on target",
+					"target", targetName,
+					"reason", event.Reason,
+					"message", event.Message,
+					"timestamp", event.LastTimestamp.Time.Unix(),
+				)
+			}
+		}
+	}
+
+	return eventsToSend, cannotRecoverYet
+}
+
+func (d DisruptionTargetHandler) buildPodEventsToSend(oldPod v1.Pod, newPod v1.Pod, disruptionEvents []v1.Event) map[string]bool {
+	var recoverTimestamp *time.Time // keep track of the timestamp of a recovering event / state
+
+	eventsToSend := make(map[string]bool)
+	runningState := "Running"
+
+	// compare statuses between old and new pod to detect changes
+	for _, container := range newPod.Status.ContainerStatuses {
+		for _, oldContainer := range oldPod.Status.ContainerStatuses {
+			if container.Name != oldContainer.Name { // When restarting, container can change of ID, so we need to verify by name
```

Comment:
It's also possible for the name to change, but I think that's rare?

Reply:
FAK: 
- Deployment: If the name of the container is updated a new replicaset is created and a new pod is scheduled. If we have a new pod, is it considered as a new target? If yes, I think it will not be a problem.
- Node: I don't know the behavior.